File: leetcode/algorithms/easy/two-sum-ii-input-array-is-sorted.js
/*
    leetcode
    category: "Algorithms"
    problem title: "Two Sum II - Input Array is Sorted"
*/

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let low = 0, high = numbers.length-1;
    for(let i=0;i<Math.floor(numbers.length/2);i++){
        let comp = target-numbers[low];
        let j;
        for(j=high;numbers[j]>comp;j--);
        high=j;
        if(numbers[high]===comp){
            return [low+1,high+1];
        }
        comp = target-numbers[high];
        for(j=low;numbers[j]<comp;j++);
        low=j;
        if(numbers[low]===comp){
            return [low+1,high+1];
        }
    }
    
};

File: leetcode/algorithms/two-ptr/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0,h = numbers.length-1;
        
        int[] sol = new int[]{-1,-1};
        
        while(l<h){
            int currSum = numbers[l] + numbers[h];
            if(currSum==target){
                sol[0] = l+1;
                sol[1] = h+1;
                break;
            } else if(currSum<target){
                ++l;
            } else {
                --h;
            }
        }
        
        return sol;
    }
}